packages/flutter/lib/src/widgets/text.dart

```diff
@@ -825,7 +787,76 @@ class _SelectableTextContainerState extends State<_SelectableTextContainer> {
   Widget build(BuildContext context) {
     return SelectionContainer(
       delegate: _selectionDelegate,
-      child: widget.child,
+      // Use [_RichText] wrapper so the underlying [RenderParagraph] can register
+      // its [Selectable]s to the [SelectionContainer] created by this widget.
+      child: _RichText(
```

Comment:
Nit: Would it be any better to use a Builder here instead of creating this _RichText widget?

Reply:
Talked offline, going to use `_RichText` widget for now. I'm not sure which one is better, but I know regarding a similar topic in helper methods vs widgets we recommend widgets https://www.youtube.com/watch?v=IOyq-eTRhvo&t=311 .